pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -376,6 +376,11 @@ func (plugin *cniNetworkPlugin) buildCNIRuntimeConf(podName string, podNs string
 		},
 	}
 
+	//send the annotations as extra args to the cni plugins
+	for key, value := range annotations {
+		rt.Args = append(rt.Args, [2]string{key, value})
```

Comment:
It's dangerous to append pod annotations to Args, because Args is passed to CNI plugin through environment variable, it is [a unified string](https://github.com/containernetworking/cni/blob/8c6c47d1c7fcf51c8d0c939d2af06dd108f876da/pkg/invoke/args.go#L77) like "A=1,B=2" but not a json format string, we have no idea about pod annotation which can include "=" or "," , so it is dangerous.

Reply:
I actually wanted to send them as args instead of CNI_ARGS but was confused a little bit. Updated the PR with the new changes which shouldn't have the issue you mentioned